middlewares/auth.js

```diff
@@ -0,0 +1,21 @@
+const authDao = require("../models/authDao");
+const jwt = require("jsonwebtoken");
+const ErrorCreater = require("../middlewares/errorCreater");
+
+const validationToken = async (req, res, next) => {
+ 
+    const accessToken = req.headers.authorization;
+    const decode = jwt.verify(accessToken, process.env.JWT_SECRET);
+    const userId = decode.sub;
+    const user = await authDao.getUserById(userId);
+
+    if (!user) {
+      throw new ErrorCreater("USER_NOT_DEFINED",404)
+    }else{
+      req.userId = userId;
+      next();
+    }
```

Comment:
```suggestion
    if (!user) {
      throw new ErrorCreater("USER_NOT_DEFINED",404)
    }
     
    req.user = user;
    next();
    }
```
user의 다른 정보도 사용할 수 있기 때문에 user 데이터 전부를 넘겨주는 것이 좋을 것 같아요.
그리고 else가 없어도 되는 문장이라 제외시키는 것이 가독성이 더 좋을 것 같습니다.

Reply:
이 부분만 참고하고 내용 수정해서 올리겠습니다!